examples/run_CPMG.py

```diff
@@ -31,91 +35,79 @@
-    ph1_cyc = r_[0, 1, 2, 3]
-    nPhaseSteps = 4
-if not phase_cycling:
-    ph1_cyc = 0.0
-    nPhaseSteps = 1
+    ph_overall = r_[0, 1, 2, 3]
+    ph_diff = r_[0, 2]
+    ph1_cyc = array([(j + k) % 4 for k in ph_overall for j in ph_diff])
+    ph2_cyc = array([(k + 1) % 4 for k in ph_overall for j in ph_diff])
+    nPhaseSteps = len(ph_overall) * len(ph_diff)
 # }}}
-# {{{better tau
-marker = 1.0  # 10/10/22 → what is this? → pretty sure the time needed to execute the marker command
-pad_start = config_dict["tau_extra_us"] - config_dict["deadtime_us"]
-pad_end = (
-    config_dict["tau_extra_us"] - config_dict["deblank_us"] - marker
-)  # marker + deblank
-assert(pad_start > 0), "tau_extra_us must be set to more than deadtime and more than deblank!"
-assert (pad_end > 0), "tau_extra_us must be set to more than deadtime and more than deblank!"
-twice_tau_echo_us = (  # the period between 180 pulses
-    config_dict["tau_extra_us"] * 2 + config_dict["acq_time_ms"] * 1e3
-)
-# now twice_tau_echo_us/2.0 is τ_echo, so I need to subtract the extra delays
-# imposed by the ppg to determine the remaining τ that I need
-config_dict["tau_us"] = twice_tau_echo_us / 2.0 - (
-    2
-    * config_dict["p90_us"]
-    / pi  # evolution during pulse -- see eq 6 of coherence paper
-    + config_dict["deadtime_us"]  # following 90
-    + config_dict["deblank_us"]  # before 180
-)
+# {{{symmetric tau
+short_delay_us = 1.0
+tau_evol_us = (
+    2 * config_dict["p90_us"] / pi
+)  # evolution during pulse -- see eq 6 of coherence paper
+pad_end_us = config_dict["deadtime_us"] - config_dict["deblank_us"] - 2 * short_delay_us
+twice_tau_echo_us = config_dict["acq_time_ms"] + (
+    2 * config_dict["deadtime_us"]
+)  # the period between end of first 180 pulse and start of next
+config_dict["tau_us"] = twice_tau_echo_us / 2.0 - tau_evol - config_dict["deblank_us"]
 # }}}
 # {{{check total points
-total_pts = nPoints * nPhaseSteps
-assert total_pts < 2 ** 14, (
+total_pts = nPoints * nPhaseSteps * config_dict["nEchoes"]
+assert total_pts < 2**14, (
     "You are trying to acquire %d points (too many points) -- either change SW or acq time so nPoints x nPhaseSteps is less than 16384\nyou could try reducing the acq_time_ms to %f"
     % (total_pts, config_dict["acq_time_ms"] * 16384 / total_pts)
 )
 # }}}
 # {{{run cpmg
-# NOTE: Number of segments is nEchoes * nPhaseSteps
-data = run_cpmg(
-    nScans = config_dict["nScans"],
-    indirect_idx = 0,
-    indirect_len = 1,
-    ph1_cyc = ph1_cyc,
-    adcOffset = config_dict["adc_offset"],
-    carrierFreq_MHz = config_dict["carrierFreq_MHz"],
-    nPoints = nPoints,
-    nEchoes = config_dict["nEchoes"],
-    p90_us = config_dict["p90_us"],
-    repetition_us = config_dict["repetition_us"],
-    tau_us = config_dict["tau_us"],
-    SW_kHz = config_dict["SW_kHz"],
-    pad_start_us = pad_start,
-    pad_end_us = pad_end,
-    ret_data = None,
+data = generic(
+    ppg_list=[
+        ("marker", "start", 1),
```

Comment:
Is this normal? Shouldn't there be a mention of nScans here?

Reply:
line 86 below